src/main/scala/scalaz/netty/Client.scala

```diff
@@ -36,6 +36,7 @@ import _root_.io.netty.handler.codec._
 
 private[netty] final class Client(limit: Int) {
   // this isn't ugly or anything...
+  @volatile
```

Comment:
This should be small amount of work - I will do it after hours. Interface of the factory method for creating Task[Server] won't be changed, so this'd be just internal.